hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -589,15 +589,15 @@ public boolean getSafeModeStatus() {
    * Query the System for Nodes.
    *
    * @param nodeState - NodeState that we are interested in matching.
-   * @return Set of Datanodes that match the NodeState.
+   * @return List of Datanodes that match the NodeState.
    */
-  private Set<DatanodeDetails> queryNodeState(HddsProtos.NodeState nodeState) {
-    Set<DatanodeDetails> returnSet = new TreeSet<>();
+  private List<DatanodeDetails> queryNodeState(HddsProtos.NodeState nodeState) {
+    List<DatanodeDetails> returnList = new ArrayList<>();
     List<DatanodeDetails> tmp = scm.getScmNodeManager().getNodes(nodeState);
     if ((tmp != null) && (tmp.size() > 0)) {
-      returnSet.addAll(tmp);
+      returnList.addAll(tmp);
```

Comment:
I think `returnList` is no longer needed, `tmp` can be returned from here directly.